stacks-signer/src/runloop.rs

```diff
@@ -287,7 +291,12 @@ impl RunLoop {
             *reward_cycle_info = new_reward_cycle_info;
         }
         let current_reward_cycle = reward_cycle_info.reward_cycle;
-        // We should only attempt to refresh the signer if we are not configured for the next reward cycle yet and we received a new burn block for its prepare phase
+        let current_burn_block_height = reward_cycle_info.last_burnchain_block_height;
```

Comment:
Why overwrite the `current_burn_block_height` passed in as a function arg?

Reply:
beacuse our view may have changed. Most likely not. But could happen. and we already get themost current burn block info by calling get reward cycle info again, so figured, make it consistent. It most liekly is the same value, but we cannot guarantee it.

Reply:
Correct solution is to use the burn block height of the Block proposal of the signer (they do store the burn block height at which it was created) but I haven't done that yet because I want confirmation that this is the approach to go with as its slightly more involved change.

Reply:
Ended up just doing it.